persist/fs/read.go

```diff
@@ -120,22 +120,50 @@ func NewReader(
 	}, nil
 }
 
-func (r *reader) Open(namespace ident.ID, shard uint32, blockStart time.Time) error {
-	var err error
+func (r *reader) Open(opts ReaderOpenOptions) error {
+	var (
+		namespace     = opts.Namespace
+		shard         = opts.Shard
+		blockStart    = opts.BlockStart
+		isSnapshot    = opts.IsSnapshot
+		snapshotIndex = opts.SnapshotIndex
+		err           error
+	)
+
+	var shardDir string
+	var checkpointFilePath string
+	if isSnapshot {
+		shardDir = ShardSnapshotsDirPath(r.filePathPrefix, namespace, shard)
+		checkpointFilePath = snapshotPathFromTimeAndIndex(shardDir, blockStart, checkpointFileSuffix, snapshotIndex)
+	} else {
+		shardDir = ShardDataDirPath(r.filePathPrefix, namespace, shard)
+		checkpointFilePath = filesetPathFromTime(shardDir, blockStart, checkpointFileSuffix)
+	}
 
 	// If there is no checkpoint file, don't read the data files.
-	shardDir := ShardDirPath(r.filePathPrefix, namespace, shard)
-	if err := r.readCheckpointFile(shardDir, blockStart); err != nil {
+	if err := r.readCheckpointFile(checkpointFilePath, blockStart); err != nil {
 		return err
 	}
 
 	var infoFd, digestFd *os.File
-	if err := openFiles(os.Open, map[string]**os.File{
-		filesetPathFromTime(shardDir, blockStart, infoFileSuffix):        &infoFd,
-		filesetPathFromTime(shardDir, blockStart, digestFileSuffix):      &digestFd,
-		filesetPathFromTime(shardDir, blockStart, bloomFilterFileSuffix): &r.bloomFilterFd,
-	}); err != nil {
-		return err
+	if isSnapshot {
```

Comment:
Yeah if this was an enum type we could do a switch here which would be nice.

Reply:
done